test/wallet.spec.ts

```diff
@@ -132,3 +133,40 @@ describe('Wallet', () => {
         fs.rmSync('./test/wallet', { recursive: true, force: true });
     });
 });
+
+describe('Wallet - DB', () => {
+    let wallet: Wallet;
+
+    beforeAll(async () => {
+        const walletDB = new WalletDB({
+            location: './test/wallet',
+        });
+
+        wallet = new Wallet({
+            db: walletDB,
+            networkClient: new EsploraClient({
+                protocol: 'http',
+                host: '127.0.0.1:8094',
+                network: 'regtest',
+            }),
+        });
+        await wallet.init({
+            mnemonic:
+                'abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about',
+            password: 'samplePassword',
+        });
+    });
+
+    it('should write masterKey with new password', async () => {
+        await wallet.writeMasterKey('newPassword');
+        await wallet.close();
+        await wallet.init({ password: 'newPassword' });
+        const address = await wallet.deriveReceiveAddress();
+        expect(address).toBe('bcrt1qcr8te4kr609gcawutmrza0j4xv80jy8zeqchgx');
+    });
+
+    afterAll(async () => {
+        await wallet.close();
+        fs.rmSync('./test/wallet', { recursive: true, force: true });
+    });
+});
```

Comment:
Don't think we need a new describe block just to test one feature. Just use the setup I have made earlier in the file. This patch should do it.

```
diff --git a/test/wallet.spec.ts b/test/wallet.spec.ts
index 3b172f7..f1a4081 100644
--- a/test/wallet.spec.ts
+++ b/test/wallet.spec.ts
@@ -38,6 +38,13 @@ describe('Wallet', () => {
         });
     });
 
+    it('should set a new password, close and reopen the wallet with the same password', async () => {
+        const password = 'notSoSecretPassword';
+        await wallet.writeMasterKey(password);
+        await wallet.close();
+        await wallet.init({ password });
+    });
+
     it('should derive first receive address', async () => {
         address = await wallet.deriveReceiveAddress();
         expect(address).toBe('bcrt1qcr8te4kr609gcawutmrza0j4xv80jy8zeqchgx');
```

Reply:
This will not work as the `it` blocks are run concurrently and the tests executed when the wallet is closed fail. This is a flaky test and should be avoided imo.
![Screenshot 2024-02-10 at 1 29 50 PM](https://github.com/Bitshala-Incubator/silent-pay/assets/32434322/04cb347d-b210-4002-bacf-96ccc8a6ad95)

Hence I separated this out to a separate wallet instance. Maybe we can have two wallet instances in this `describe` block, but again two `describe` blocks seemed cleaner.

Reply:
The `it` blocks in a `describe` run sequentially, always. It's the `describe` that runs concurrently.

Reply:
Yes my initial RCA was wrong, this is an issue with the `wallet.close()` method. Thanks for helping with the debugging @theanmolsharma